ClothesApp/Application.UnitTests/AddressService/AddressServiceTests.cs

```diff
@@ -0,0 +1,150 @@
+            LastName = faker.Name.LastName(),
+        };
+    }
+
+    [Theory]
+    [ClassData(typeof(AddAddressTestData))]
+    public async Task AddAddress_WhereCalledWithCorrectAddAddressDto(AddAddressTestCase testCase)
+    {
+        var addressService = CreateAddressService(Context);
+        Context.Users.Add(testCase.User);
+        await Context.SaveChangesAsync();
+        
+        var act = await addressService.AddAddress(testCase.AddAddressDto);
+        
+        act.Should().BeEquivalentTo(testCase.AddAddressDto);
+    }
+    
+    [Theory]
+    [ClassData(typeof(UpdateAddressesTestData))]
+    public async Task UpdateAddress_WhereCalledWithANonRelativeUserAddress_ThrowsBusinessRule(UpdateAddressTestCase testCase)
+    {
+        var addressService = CreateAddressService(Context);
+        Context.Users.Add(testCase.User);
+        var address = testCase.Address;
+        address.UserId =- 1;
+        Context.Addresses.Add(testCase.Address);
+        await Context.SaveChangesAsync();
+        
+        Func<Task> act = async () => await addressService.UpdateAddress(testCase.User.Id, testCase.AddressInputDto);
+
+        await act.Should().ThrowAsync<BusinessRuleException>();
+    }
+    
+    [Fact]
+    public async Task Update_WhereCalledWithUndiscoveredUserId_ThrowsNotFound()
+    {
+        var addressService = CreateAddressService(Context);
+        var faker = new Faker();
+        var user = GenerateUser();
+        var addressInputDto = new AddressInputDto()
+        {
+            Id = faker.Random.Long(1),
+            AddressLine = faker.Address.FullAddress(),
+        };
+        Context.Users.Add(user);
+
+        Func<Task> act = async () => await addressService.UpdateAddress(user.Id - 1, addressInputDto);
+
+        await act.Should().ThrowAsync<NotFoundException>();
+    }
+    
+    [Fact]
+    public async Task UpdateAddress_WhereCalledWithUndiscoveredAddressId_ThrowsNotFound()
+    {
+        var addressService = CreateAddressService(Context);
+        var faker = new Faker();
+        var addressInputDto = new AddressInputDto()
+        {
+            Id = faker.Random.Long(1),
+            AddressLine = faker.Address.FullAddress(),
+        };
+        var user = GenerateUser();
+        Context.Users.Add(user);
+        await Context.SaveChangesAsync();
+
+        Func<Task> act = async () => await addressService.UpdateAddress(user.Id, addressInputDto);
+
+        await act.Should().ThrowAsync<NotFoundException>();
+    }
+    
+    [Theory]
+    [ClassData(typeof(UpdateAddressesTestData))]
+    public async Task UpdateAddress_WhereCalledWithCorrectData(UpdateAddressTestCase testCase)
+    {
+        var addressService = CreateAddressService(Context);
+        
+        Context.Users.Add(testCase.User);
+        Context.Addresses.Add(testCase.Address);
+        await Context.SaveChangesAsync();
+        
+        var act = await addressService.UpdateAddress(testCase.User.Id, testCase.AddressInputDto);
```

Comment:
When you test create/update methods, I would suggest you to assert against the context data, not against what is returned from this method. Because on create you can return whatever you want but the most important thing is how the state actually changed